### PR TITLE
Do not tolerate authentication-tolerate-lookup-failure

### PR DESCRIPTION
**Pull request by alvaroaleman**
**What this PR does / why we need it**:

Currently if the lookup of the authentication config lookup fails, the scheduler and controller-manager will just continue starting up, but treat all requests as anonymous and thus deny them. This patch makes a lookup of the authentication config a fatal error.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes the sporadic `Kubernetes scheduler has disappeared from Prometheus target discovery.`

**Special notes for your reviewer**:

**Documentation**:

**Release note**:

```release-note
none
```

### CODE REVIEW

**Comment by alvaroaleman** [comment]
cc @xrstf

File: api/pkg/resources/scheduler/deployment.go [hunk 1]
```diff
@@ -200,6 +200,8 @@ func getFlags(cluster *kubermaticv1.Cluster) ([]string, error) {
 		flags = append(flags, "--authorization-kubeconfig", "/etc/kubernetes/kubeconfig/kubeconfig")
 		// We're going to use the https endpoints for scraping the metrics starting from 1.13. Thus we can deactivate the http endpoint
 		flags = append(flags, "--port", "0")
+		// Force the authentication lookup to succeed, otherwise if it fails all requests will be treated as anonymous and thus fail
+		flags = append(flags, "--authentication-tolerate-lookup-failure", "false")
```
Comment (mrIncompetent) [inline api/pkg/resources/scheduler/deployment.go]: This flag does not exist in v1.13.0.
It got introduced as a Bugfix with https://github.com/kubernetes/kubernetes/pull/71806

But we can simply omit the flag(for v1.13.0) as failing was the default behavior.

**Review by mrIncompetent — CHANGES_REQUESTED** [review]

File: api/pkg/resources/scheduler/deployment.go (hunk 1 above)
Comment (alvaroaleman) [reply api/pkg/resources/scheduler/deployment.go]: Wow, that is a detailled knowledge of the releases. Will fix it up.

Comment (alvaroaleman) [reply api/pkg/resources/scheduler/deployment.go]: Updated, PTAL

**Review by mrIncompetent — APPROVED** [review]

**Pull request merged**